UMFPack directory not handed to applications

I am using an external installation of UMFPack and deal.II configures and compiles correctly using
```
UMFPACK_INCLUDE_DIR              /usr/include/suitesparse
```
I am initializing my application with
```
DEAL_II_INITIALIZE_CACHED_VARIABLES()
DEAL_II_SETUP_TARGET(amandus)
```
The include directory is not forwarded and eventually my compiler fails:
```
/home/guido/deal/dealii/include/deal.II/lac/sparse_direct.h:30:23: fatal error: umfpack.h: No such file or directory
 #  include <umfpack.h>
```
Should all include directories be set in the compiler flags for applications or should I configure UMFPack myself?

Best,
Guido

> /home/guido/deal/dealii/include/deal.II/lac/sparse_direct.h:30:23: fatal error: umfpack.h: No such file or directory

This regression was introduced with 053bb824000d7aeb0edb37d125eff7a9e417bc8a 

> Should all include directories be set in the compiler flags for
> applications or should I configure UMFPack myself?

Prior to above merge it was not necessary to directly include suite
sparse headers, thus they were not present in the project configuration.
A fix is:

diff --git a/cmake/modules/FindUMFPACK.cmake b/cmake/modules/FindUMFPACK.cmake
index 88e439c..bd26c93 100644
--- a/cmake/modules/FindUMFPACK.cmake
+++ b/cmake/modules/FindUMFPACK.cmake
@@ -1,6 +1,6 @@
 ## ---------------------------------------------------------------------
 ##
-## Copyright (C) 2012 - 2014 by the deal.II authors
+## Copyright (C) 2012 - 2015 by the deal.II authors
 ##
 ## This file is part of the deal.II library.
 ##
@@ -154,6 +154,9 @@ DEAL_II_PACKAGE_HANDLE(UMFPACK
   INCLUDE_DIRS
     REQUIRED UMFPACK_INCLUDE_DIR AMD_INCLUDE_DIR
     OPTIONAL SuiteSparse_config_INCLUDE_DIR
+  USER_INCLUDE_DIRS
+    REQUIRED UMFPACK_INCLUDE_DIR AMD_INCLUDE_DIR
+    OPTIONAL SuiteSparse_config_INCLUDE_DIR
   LINKER_FLAGS
     OPTIONAL LAPACK_LINKER_FLAGS
   CLEAR


Best,
Matthias

--- a/cmake/modules/FindUMFPACK.cmake
+++ b/cmake/modules/FindUMFPACK.cmake
@@ -1,6 +1,6 @@
 ## ---------------------------------------------------------------------
 ##
-## Copyright (C) 2012 - 2014 by the deal.II authors
+## Copyright (C) 2012 - 2015 by the deal.II authors
 ##
 ## This file is part of the deal.II library.
 ##
@@ -152,6 +152,9 @@ DEAL_II_PACKAGE_HANDLE(UMFPACK
     REQUIRED AMD_LIBRARY
     OPTIONAL METIS_LIBRARIES LAPACK_LIBRARIES rt_LIBRARY
   INCLUDE_DIRS
+    REQUIRED UMFPACK_INCLUDE_DIR AMD_INCLUDE_DIR
+    OPTIONAL SuiteSparse_config_INCLUDE_DIR
+  USER_INCLUDE_DIRS
     REQUIRED UMFPACK_INCLUDE_DIR AMD_INCLUDE_DIR
     OPTIONAL SuiteSparse_config_INCLUDE_DIR
   LINKER_FLAGS

